Sources/Attachment.php

```diff
@@ -1469,7 +1469,7 @@ public static function create(&$attachmentOptions)
 			);
 		}
 
-		if (!$this->is_image || empty(Config::$modSettings['attachmentThumbnails']) || (empty($attachmentOptions['width']) && empty($attachmentOptions['height']))) {
+		if (empty($image) || empty(Config::$modSettings['attachmentThumbnails']) || (empty($attachmentOptions['width']) && empty($attachmentOptions['height']))) {
```

Comment:
This won't quite work, because $image is always unset on line 1346 above.

But you could just add `$is_image = true;` inside the if block that starts on line 1321, and then change this line here to use `empty($is_image)`.